### DISCUSSION TITLE
Clearing State with JS SDK

### DISCUSSION THREAD

**Discussion by BenKurrek** [near/near-sdk-js#27] [Q&A] [answered]
I think it would be neat to have a `near clear-state` command because clearing your state using the JS SDK is not as simple as it is with Rust or AS. In Rust and AS, you could just delete the sub-account and redeploy which is a very common dev approach and we’ve been pushing people towards doing that for a while. With the JS SDK, you have to deploy a separate contract and call some methods on that to clear the state. Maybe we can abstract that into a simple function that they call which does all that for them to avoid some potential pain points.

**Comment by austinbaggio** [comment]
@ChaoticTempest We should add this to the CLI discussion https://github.com/near/near-cli/discussions/978

**Comment by ChaoticTempest** [comment]
> you could just delete the sub-account and redeploy which is a very common dev approach

This is kind of workflow is already available via the CLI:
```
near js deploy --accountId account.testnet
near js remove --accountId account.testnet
```

**Reply by volovyks** [reply]
@ChaoticTempest does it clear the contract state? Or deletes the contract itself?

**Reply by ChaoticTempest** [reply]
I'm assuming here that `remove_js_contract` also deletes the state since the deposit is refunded. It's the equivalent of deleting the contract account, except the actual account isn't deleted, just the state in the enclave.

**Reply by volovyks** [reply]
Hmm... if we can delete the state with one command, should we have `near-sdk-js/examples/clean-state` example at all?
@BenKurrek @ailisp maybe I'm losing something here.

**Comment by ailisp** [comment, marked as answer]
We've had a clear state example and usage equivalent to Rust's, thanks for @itegulov :
https://github.com/near/near-sdk-js/tree/master/examples/clean-state

**Reply by BenKurrek** [reply]
While this doesn't allow you to clear state by deleting and recreating the account similar to native NEAR, this will do.
